npm/templates/deploy.py

```diff
@@ -66,19 +67,11 @@
         '$DEPLOY_NPM_EMAIL env variable'
     )
 
-expect_input_tmpl = '''spawn npm adduser --registry={registry}
-expect {{
-  "Username:" {{send "{username}\r"; exp_continue}}
-  "Password:" {{send "$env(PASSWORD)\r"; exp_continue}}
-  "Email: (this IS public)" {{send "{email}\r"; exp_continue}}
-}}'''
-
-with tempfile.NamedTemporaryFile('wt', delete=False) as expect_input_file:
-    expect_input_file.write(expect_input_tmpl.format(
-        registry=npm_registry,
-        username=npm_username,
-        email=npm_email,
-    ))
+if repo_type == "release" and not npm_token:
+    print(
+        'token should be passed via '
+        '$DEPLOY_NPM_TOKEN env variable when deploying to a release repo'
+    )
```

Comment:
`DEPLOY_NPM_TOKEN` must now be provided when deploying to a release repo (note: this is just a sanity check. Release repos typically require authentication and snapshot repos typically don't)

Reply:
Are we sure we don't have to provide authentication credential when deploying to the snapshot repo? Doesn't that mean anyone can deploy a snapshot artefact if they know the URL?

Reply:
It's a sufficient (and not necessary) condition to provide $DEPLOY_NPM_TOKEN to a snapshot repo. My thought was that some packages in snapshot repos might not require authentication to publish.

But to be honest I've never seen an example of this, so I'll update the check to make DEPLOY_NPM_TOKEN always mandatory.